docs/dub-reference/dependencies.md

```diff
@@ -0,0 +1,25 @@
+# Dependency syntax
```

Comment:
Header should be `Dependencies`, since it's what this file is called. Create a sub-heading for the syntax, link to `build_settings.md#dependencies`

Also link to the draft / planning issue here on github while this page isn't complete.